algebras/algebra/src/test/scala/endpoints/algebra/server/EndpointsTestSuite.scala

```diff
@@ -0,0 +1,29 @@
+package endpoints.algebra.server
+import com.softwaremill.sttp.quick._
+import endpoints.algebra.EndpointsTestApi
+
+trait EndpointsTestSuite[T <: EndpointsTestApi] extends ServerTestBase[T] {
+
+  def serverTestSuite() = {
+
+    "Server interpreter" should {
+
+      "return server response" in {
+
+        val mockedResponse = "interpretedServerResponse"
+
+        val server = serveEndpoint(serverApi.smokeEndpoint, mockedResponse)
+        server.start()
+
+        val response  = sttp.get(uri"http://localhost:${server.port}/user/userId/description?name=name1&age=18").send()
+        assert(response.body.isRight)
+        assert(response.body.right.get == mockedResponse)
+        assert(response.code == 200)
+
+        server.stop()
+      }
```

Comment:
What do you think if, instead, `serveEndpoint` takes an additional function parameter that is called after the server is started and before it is stopped? Test writers wouldn’t have to manually take care of starting and stopping the server.